### PR TITLE
Allow option to force destroy of s3 bucket, default being false

### LINKED ISSUE #193: Optional S3 bucket force_destroy
### Persona

The Defense Unicorns Big Bang Distribution (DUBBD) CI pipeline utilizes this repo as a module in Zarf components to stand up AWS resources needed for a Big Bang deployment on EKS, namely an S3 bucket.

### Description

The desired outcome would be to allow some means of destroying this S3 bucket forcefully.

### Use Case

As part of the CI process that validates code changes to DUBBD, it does a net new environment creation (EKS, AWS pieces, deploy DUBBD) and then destroys all those pieces.  The DUBBD CI process was unable to complete the destroy step as the S3 bucket created from this repo's S3 module was getting this error:

```
│ Error: deleting Amazon S3 (Simple Storage) Bucket (big-bang-dfe4fd420230421123932518900000001): BucketNotEmpty: The bucket you tried to delete is not empty. You must delete all versions in the bucket.
│ 
     ERROR:  Unable to remove the package with an error of: &fmt.wrapError{msg:"unable to remove the component
 (execute-terraform): unable to run the before action for component (execute-terraform): command
 \"terraform destroy -auto-approve\" failed after 0 retries", err:(*fmt.wrapError)(0xc001a389c0)}
Error: Process completed with exit code 1.
```

### Impact

The DUBBD pipeline validation fails which blocks PRs from being merged. 

### Completion

* The S3 module in this repo would allow an option to be passed to allow for a "force_destroy"
* This option would default to "false" so that it doesn't impact existing usage of this module
* The S3 bucket used by the DUBBD CI pipeline deletes successfully on the destroy step

### PR DESCRIPTION
**Pull request by MxNxPx**
This PR addresses Issue: https://github.com/defenseunicorns/delivery-aws-iac/issues/193

It has been validated in DUBBD by pinning to this branch: https://github.com/defenseunicorns/zarf-package-big-bang/blob/main/aws/loki/main.tf#L9

### CODE REVIEW

**Review by RothAndrew — CHANGES_REQUESTED** [review]

File: modules/s3-irsa/variables.tf [hunk 1]
```diff
@@ -103,3 +103,9 @@ variable "access_logging_bucket_prefix" {
   type        = string
   default     = "s3-irsa-bucket-access-logs/"
 }
+
+variable "force_destroy" {
+  description = "Option to set force destroy"
```
[the remark this answers is not in the document]
Comment (MxNxPx) [reply modules/s3-irsa/variables.tf]: thanks @RothAndrew, good feedback!

**Comment by RothAndrew** [comment]
The pre-commit hooks need to be run, then this should be ready to go.

**Comment by RothAndrew** [comment]
@MxNxPx please run `make run-pre-commit-hooks` and push up the changes it makes.

**Review by RothAndrew — APPROVED** [review]

**Pull request merged**